src/EditorFeatures/CSharpTest/ConvertAutoPropertyToFullProperty/ConvertAutoPropertyToFullPropertyTests.cs

```diff
@@ -0,0 +1,965 @@
+
+class MyDerivedClass : MyBaseClass
+{
+    public override string Name {get; set;}
+}
+";
+            var expected = @"
+class MyBaseClass
+{
+    private string _name;
+
+    public virtual string Name
+    {
+        get
+        {
+            return _name;
+        }
+        set
+        {
+            _name = value;
+        }
+    }
+}
+
+class MyDerivedClass : MyBaseClass
+{
+    public override string Name {get; set;}
+}
+";
+            await TestInRegularAndScriptAsync(text, expected, options: DoNotPreferExpressionBodiedAccessors);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ConvertAutoPropertyToFullProperty)]
+        public async Task PrivateProperty()
+        {
+            var text = @"
+class MyClass
+{
+    private string N[||]ame { get; set; }
+}
+";
+            var expected = @"
+class MyClass
+{
+    private string _name;
+
+    private string Name
+    {
+        get
+        {
+            return _name;
+        }
+        set
+        {
+            _name = value;
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected, options: DoNotPreferExpressionBodiedAccessors);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ConvertAutoPropertyToFullProperty)]
+        public async Task AbstractProperty()
+        {
+            var text = @"
+class MyBaseClass
+{
+    public abstract string N[||]ame { get; set; }
+}
+
+class MyDerivedClass : MyBaseClass
+{
+    public override string Name {get; set;}
+}
+";
+            await TestMissingAsync(text);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ConvertAutoPropertyToFullProperty)]
+        public async Task ExternProperty()
+        {
+            var text = @"
+class MyBaseClass
+{
+    extern string N[||]ame { get; set; }
+}
+";
+            await TestMissingAsync(text);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ConvertAutoPropertyToFullProperty)]
+        public async Task GetterOnly()
+        {
+            var text = @"
+class goo
+{
+    public int G[||]oo { get;}
+}
+";
+            var expected = @"
+class goo
+{
+    private readonly int _goo;
+
+    public int Goo
+    {
+        get
+        {
+            return _goo;
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected, options: DoNotPreferExpressionBodiedAccessors);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ConvertAutoPropertyToFullProperty)]
+        public async Task SetterOnly()
+        {
+            var text = @"
+class goo
+{
+    public int G[||]oo
+````{
+        set {}
+    }
+}
+";
+            await TestMissingAsync(text);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ConvertAutoPropertyToFullProperty)]
+        public async Task ExpressionBodiedAccessors()
+        {
+            var text = @"
+class goo
+{
+   private int _testgoo;
+
+   public int testf[||]oo {get => _testgoo; set => _testgoo = value; }
```

Comment:
do not use foo.

Reply:
Do you have a test where we start with "int Prop { get; }" and the user wants "expression bodied properties", so we conver that to "int Prop => _prop"?    If not, can you add it?

Reply:
Added this test